### DISCUSSION TITLE
LFP Cell Volatge too Low and discharge Capacity too High

### DISCUSSION THREAD

**Discussion by pmotakef** [pybamm-team/PyBaMM#2235] [Q&A]
Hi, 
Sorry for the dumb question, I just started using PyBaMM and I don't know many things. But something looks odd to me. I am using Prada2013 parameters set to simulate LFP cell, but with 1C charging the voltage never reaches 3.6 V and also during discharge if I set the cut off voltage to 2V then I get the error that electrolyte depleted and can't reach the 2V. Also, another odd thing is that when I plot the Depth of discharge it goes to 1.6 Ah while the nominal capacity is 1.1 Ah. So, I was wondering what is going wrong in here.

![image](https://user-images.githubusercontent.com/9141480/184892145-37e26188-c955-48ad-baeb-51d44aa48657.png)

Thanks a lot.

**Comment by brosaplanella** [comment]
Can you post your code snippet so we can see exactly how you define the experiment?

**Reply by pmotakef** [reply]
Sure, here is the code I used to get the graphs above (experiment 2):

```
import pybamm
import numpy as np

experiment1 = pybamm.Experiment([
    ("Charge at 6 C for 5 minutes",
     "Charge at 3.6 C for 5 minutes",
     "Rest for 1 minute",
     "Charge at 1C until 3.6V",
     "Hold at 3.6V until C/50",
     "Discharge at 4C until 2.2 V",
     "Hold at 2.2V until C/20",
     "Rest for 10 minute",
    ),
] * 100)

experiment2 = pybamm.Experiment([
    ("Charge at 1C until 3.6V",
     "Hold at 3.6V until C/50",
     "Discharge at 1C until 2.2 V",
     "Hold at 2.2V until C/20",
    ) * 1,
], period="10 seconds")

chemistry = pybamm.parameter_sets.Prada2013
chemistry['sei'] = 'example'
chemistry['lithium plating'] = 'yang2017_Li_plating'

parameter_values = pybamm.ParameterValues(chemistry=chemistry)
options = {"SEI": "ec reaction limited", "calculate discharge energy": 'true'}
model = pybamm.lithium_ion.DFN(options=options)

sim = pybamm.Simulation(model, parameter_values=parameter_values, experiment=experiment2)
sim.solve(initial_soc=0.01)
sim.plot(["Current [A]", "Terminal voltage [V]", "Discharge capacity [A.h]"])
```

Experiment 1 was designed to replicate the results of this paper: https://data.matr.io/1/projects/5c48dd2bc625d700019f3204

Thanks

**Comment by brosaplanella** [comment]
I am not very familiar with this parameter set, but the values come from various sources. It could be that this is why you can't reproduce the results you want.

**Reply by pmotakef** [reply]
Yes it might be because of the dataset. I was wondering if there is guide for parameters and how they influence the results? That would be helpful if I know which parameters to look and try to tune.
I thought since in the dataset the capacity is set to 1.1 A.h, in the simulation we shouldn't see values beyond that range. But, I am unfamiliar with the model, so maybe this is expected.
```
Nominal cell capacity [A.h]	1.1
```

Another thing that I don't understand is why the solver goes beyond the bounds. For example in experiment 1, I am getting this error after 78 cycles (my cut off voltage at charging is set to 3.0 V instead of 2.2 V in this example):

```
2022-08-16 14:26:24.481 - [NOTICE] callbacks.on_step_start(182): Cycle 78/100, step 1/8: Charge at 6 C for 5 minutes
2022-08-16 14:26:24.621 - [NOTICE] callbacks.on_step_start(182): Cycle 78/100, step 2/8: Charge at 3.6 C for 5 minutes
2022-08-16 14:26:24.757 - [NOTICE] callbacks.on_step_start(182): Cycle 78/100, step 3/8: Rest for 1 minute
2022-08-16 14:26:24.818 - [NOTICE] callbacks.on_step_start(182): Cycle 78/100, step 4/8: Charge at 1C until 3.5V
2022-08-16 14:26:25.312 - [NOTICE] callbacks.on_step_start(182): Cycle 78/100, step 5/8: Hold at 3.5V until C/50
2022-08-16 14:26:25.455 - [NOTICE] callbacks.on_step_start(182): Cycle 78/100, step 6/8: Discharge at 4C until 3.0 V
2022-08-16 14:26:25.908 - [WARNING] callbacks.on_experiment_infeasible(237): 

	Experiment is infeasible: 'event: Zero electrolyte concentration cut-off' was triggered during 'Discharge at 4C until 3.0 V'. The returned solution only contains the first 78 cycles, up to step 5. Try reducing the current, shortening the time interval, or reducing the period.
```
![image](https://user-images.githubusercontent.com/9141480/185985456-bec840dc-fe27-4ee1-bb83-79d09c93bc2f.png)

For example the last cycle, the voltage falls below 3.0 V and the solver doesn't capture it and goes into infeasible region. Is there a parameter I should set to prevent this behavior?

Thanks a lot,

**Comment by brosaplanella** [comment]
The parameter `Nominal cell capacity [A.h]` is only used to determine the C-rate, so in fact the actual capacities can be different. 

Note that you are also using an SEI model, which means an SEI dataset is brought in and it might not be consistent with the remaining parameters. In this case, as you are getting electrolyte depletion, the parameter that might affect your results the most is the porosity. However, if you are trying to reproduce the results from a given modelling paper, it would be better to use the same parameter set used in that paper.
